src/network/clientpackethandler.cpp

```diff
@@ -217,9 +217,30 @@ void Client::handleCommand_AccessDenied(NetworkPacket* pkt)
 		*pkt >> denyCode;
 		if (denyCode == SERVER_ACCESSDENIED_CUSTOM_STRING) {
 			*pkt >> m_access_denied_reason;
-		}
-		else if (denyCode < SERVER_ACCESSDENIED_MAX) {
-			m_access_denied_reason = accessDeniedStrings[denyCode];
+		} else {
+			if (denyCode < SERVER_ACCESSDENIED_MAX) {
+				m_access_denied_reason = accessDeniedStrings[denyCode];
+				if (denyCode == SERVER_ACCESSDENIED_SHUTDOWN ||
+						denyCode == SERVER_ACCESSDENIED_CRASH) {
+					*pkt >> m_access_denied_reason;
+				} else {
+					std::string reason_str;
+					*pkt >> reason_str;
+				}
+
+				u8 reconnect = 0;
+				*pkt >> reconnect;
```

Comment:
Warn: you extract reconnect field for each SERVER_ACCESSDENIED reason but only SHUTDOWN and CRASH have this field. Client will crash here.